hdfs-plugins/src/main/java/co/cask/hydrator/plugin/HDFSSink.java

```diff
@@ -139,10 +140,17 @@ public HDFSSinkConfig(String referenceName, String path, String suffix, String o
       this.timeSufix = suffix;
     }
 
-    private void validate() {
-      // if macro provided, timeSuffix will be null at configure time
-      if (!Strings.isNullOrEmpty(timeSufix)) {
-        new SimpleDateFormat(timeSufix);
+    private void validateProperties() {
+      validateProperty("timeSuffix");
+    }
+
+    private void validateProperty(String property) {
+      switch (property) {
+        case "timeSuffix":
```

Comment:
for cases like this when there is just a single property, we don't need a switch case.